chia/wallet/wallet_singleton_store.py

```diff
@@ -0,0 +1,281 @@
+    created with foreign_keys=True
+    """
+
+    db_wrapper: DBWrapper2
+
+    @classmethod
+    async def create(cls: Type[_T_WalletSingletonStore], wrapper: DBWrapper2) -> _T_WalletSingletonStore:
+        self = cls()
+        self.db_wrapper = wrapper
+
+        async with self.db_wrapper.writer_maybe_transaction() as conn:
+            await conn.execute(
+                (
+                    "CREATE TABLE IF NOT EXISTS singletons("
+                    "coin_id blob PRIMARY KEY,"
+                    " parent_coin_info text,"
+                    " puzzle_hash text,"
+                    " amount blob,"
+                    " singleton_id blob,"
+                    " wallet_id int,"
+                    " inner_puzzle blob,"
+                    " inner_puzzle_hash blob,"
+                    " confirmed tinyint,"
+                    " confirmed_at_height int,"
+                    " spent_height int,"
+                    " lineage_proof blob,"
+                    " generation int,"
+                    " timestamp int)"
+                )
+            )
+
+            await conn.execute("CREATE INDEX IF NOT EXISTS singleton_id_index ON singletons(singleton_id)")
+            await conn.execute("CREATE INDEX IF NOT EXISTS wallet_id_index ON singletons(wallet_id)")
+
+            await conn.execute(
+                "CREATE INDEX IF NOT EXISTS confirmed_at_height_index " "ON singletons(confirmed_at_height)"
+            )
+
+            await conn.execute(
+                (
+                    "CREATE TABLE IF NOT EXISTS singleton_data("
+                    "coin_id blob,"
+                    "field_name text,"
+                    "value blob,"
+                    "FOREIGN KEY(coin_id) REFERENCES singletons(coin_id) ON DELETE CASCADE)"
+                )
+            )
+
+        return self
+
+    async def _clear_database(self) -> None:
+        async with self.db_wrapper.writer_maybe_transaction() as conn:
+            await (await conn.execute("DELETE FROM singletons")).close()
+
+    async def add_singleton_record(self, record: SingletonCoinRecord) -> None:
+        """
+        Store SingletonCoinRecord in DB
+        """
+        async with self.db_wrapper.writer_maybe_transaction() as conn:
+            columns = (
+                "coin_id, parent_coin_info, puzzle_hash, amount, singleton_id, wallet_id, "
+                "inner_puzzle, inner_puzzle_hash, confirmed, confirmed_at_height, spent_height, "
+                "lineage_proof, generation, timestamp"
+            )
+            value_placeholders = ",".join(["?"] * len(columns.split(",")))
+            await conn.execute(
+                f"INSERT or REPLACE INTO singletons ({columns}) VALUES({value_placeholders})",
+                (
+                    record.coin_id(),
+                    record.coin.parent_coin_info,
+                    record.coin.puzzle_hash,
+                    record.coin.amount,
```

Comment:
`amount` is a `uint64` and has to be stored as a binary blob (of 8 bytes), because the internal integer type in sqlite is `int64`. I'm pretty sure you have to convert it to bytes here